src/App.tsx

```diff
@@ -3,33 +3,50 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query:string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.FormEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.currentTarget.value,
+    });
+  };
 
   render() {
+    const visibleMovies = [...moviesFromServer].filter(movie => (
+      movie.title.toLocaleLowerCase().includes(this.state.query.toLocaleLowerCase())
+      || movie.description.toLocaleLowerCase().includes(this.state.query.toLocaleLowerCase())
+    ));
```

Comment:
Let's take off this logic to a separate function `getVisibleMovies()`. 
Check @MaksimusCRS comment about variables, it's good advice